app/src/Elements/LineIntersection.tsx

```diff
@@ -0,0 +1,167 @@
+import Element from './Element';
+import Point from './Point';
+import Segment from './Segment';
+import Line from './Line';
+import RectangleView from '../Components/Canvas/RectangleView';
+import { createAccordion, createAccordionItem } from '../Utils/generators';
+import { error } from '../Utils/utils';
+
+type IntersectionObject = Point | Segment | Line;
+
+type ExtendedIntersectionObject = IntersectionObject | [Point, Point];
+
+type IntersectionType = 'empty' | 'contact' | 'intersection';
+
+export default class LineIntersection extends Element {
+  readonly objects: Array<IntersectionObject>;
+  readonly point: Point | undefined;
+  readonly intersectionType: IntersectionType;
+
+  private constructor(
+    objects: Array<IntersectionObject>,
+    intersectionType: IntersectionType,
+    point: Point | undefined,
+  ) {
+    super('line_intersection');
+    this.objects = objects;
+    this.point = point;
+    this.intersectionType = intersectionType;
+
+    if (this.point) {
+      this.point.color = this.color;
+    }
+  }
+
+  private static _getIntersection(
+    objects: Array<IntersectionObject>,
+  ): Point | undefined {
+    let point = undefined;
+    const element = objects[0];
+
+    let getIntersection;
+    switch (element.type) {
+      case 'point':
+        getIntersection = (obj: IntersectionObject): Point | undefined => {
+          switch (obj.type) {
+            case 'point':
+              return (obj as Point).equals(element as Point)
+                ? (obj as Point)
+                : undefined;
+            case 'segment':
+              return (obj as Segment).intersectWithPoint(element as Point);
+            case 'line':
+              return (obj as Line).intersectWithPoint(element as Point);
+          }
+        };
+        break;
+      case 'segment':
+        getIntersection = (obj: IntersectionObject): Point | undefined => {
+          switch (obj.type) {
+            case 'point':
+              return (element as Segment).intersectWithPoint(obj as Point);
+            case 'segment':
+              return (obj as Segment).intersectWithSegment(element as Segment);
+            case 'line':
+              return (obj as Line).intersectWithSegment(element as Segment);
+          }
+        };
+        break;
+      case 'line':
+        getIntersection = (obj: IntersectionObject): Point | undefined => {
+          switch (obj.type) {
+            case 'point':
+              return (element as Line).intersectWithPoint(obj as Point);
+            case 'segment':
+              return (element as Line).intersectWithSegment(obj as Segment);
+            case 'line':
+              return (obj as Line).intersectWithLine(element as Line);
+          }
+        };
+        break;
+      default:
+        return undefined;
+    }
+
+    for (let i = 1; i < objects.length; i++) {
```

Comment:
i = 1 (not 0) ??

Reply:
`getIntersection` is a function for a search of the intersection of `objects[0]` and another object. So, why we need to intersect `objects[0]` with `objects[0]`? In addition, the result of that intersection exists only when `objects[0]` is the point.

Reply:
Ok, understand